src/contracts/middleware/OBaseMiddlewareReader.sol

```diff
@@ -746,11 +746,27 @@ contract OBaseMiddlewareReader is
         }
 
         uint256 cacheIndex = cache.epochToCacheIndex[epoch];
-        uint256 pendingOperatorsToCache = operatorsLength_ - cacheIndex;
 
+        IMiddleware.CachingState cacheState = cache.epochToNextExpectedCacheCommand[epoch];
+
+        uint256 pendingOperatorsToCache = operatorsLength_ - cacheIndex;
         // Check if cache is still not filled with the current epoch validators
         if (pendingOperatorsToCache > 0) {
             uint256 maxNumOperatorsToCheck = Math.min(pendingOperatorsToCache, MAX_OPERATORS_TO_PROCESS);
+            if (cacheState == IMiddleware.CachingState.ExpectingVaultData) {
+                (IMiddleware.VaultPower[] memory vaultsPower, bool atLeastOne, uint256 operatorsProcessed) =
+                _getOperatorPowerPerVault(
+                    maxNumOperatorsToCheck, cacheIndex, currentEpochStartTs, operators, operatorsLength_
+                );
+
+                // This should be useless since it is always checked on the validator operators process. Impossible to reach this state
```

Comment:
I think it's reachable.
1. If `!atLeastOne` that means all operators in this batch had no power. 
2. If `operatorsLength_ <= MAX_OPERATORS_TO_SEND` that means they all fit in one batch.
3. So we have all operators here, and all have no power. We can do an early exit since there's no point on sending anything to substrate. But that opens a question: 

What should we tell substrate if no operators have power? 😨